### PR TITLE
feat(rust, python): Convert `Time` and `Datetime` to number of minutes/ seconds in the day

### PR DESCRIPTION
**Pull request by ohines**
See discussion in #10593

### CODE REVIEW

File: py-polars/polars/series/datetime.py [hunk 1]
```diff
@@ -944,6 +944,82 @@ def nanosecond(self) -> Series:
...
+                2001-01-01 00:02:00
+                2001-01-01 00:04:00
+        ]
+        >>> date.dt.minute_of_day()
+        shape: (3,)
+        Series: 'date' [u32]
+        [
+                0
+                2
+                64
```
Comment (MarcoGorelli) [inline py-polars/polars/series/datetime.py:980]: hi - could you clarify how `2001-01-01 00:04:00` corresponds to `minute of the day` being `64`? Not sure I follow

File: py-polars/polars/series/datetime.py [hunk 2]
```diff
@@ -944,6 +944,82 @@ def nanosecond(self) -> Series:
...
+                2001-01-01 01:00:00
+        ]
+        >>> date.dt.second_of_day()
+        shape: (3,)
+        Series: 'date' [u32]
+        [
+                10
+                1440
+                3600
+        ]
```
Comment (MarcoGorelli) [inline py-polars/polars/series/datetime.py:1019]: is this not already possible (and efficient) with
```python
In [20]: date.dt.second() + date.dt.minute()*60 + date.dt.hour()*3600
Out[20]:
shape: (3,)
Series: 'date' [u32]
[
        10
        140
        3600
]
```
?

In addition to being concerned about "API bloat", I think this is a bit more explicit - is "second of day" a standard term? To me at least it's not obvious what it means, just by reading it

File: py-polars/polars/series/datetime.py (hunk 1 above)
Comment (ohines) [reply py-polars/polars/series/datetime.py:980]: Apologies this was a typo - the example was meant to be `2001-01-01 01:04:00`

File: py-polars/polars/series/datetime.py (hunk 2 above)
Comment (ohines) [reply py-polars/polars/series/datetime.py:1019]: I agree that it is possible and efficient but I think it is a common enough operation that a short hand function could be useful. Also it makes it easier when the date/ time is parsed directly from a string e.g
```python
time_string.str.strptime(pl.Time, "%H:%M:%S").dt.second_of_day()
```
>is "second of day" a standard term?
Not that I know of but happy to hear other suggestions

**Comment by ohines** [comment]
Closing due to concerns about API bloat.

**Pull request closed without merging**